File: ios/Classes/LogUtils.swift
import Foundation
import os.log

/**
 * 日志工具类，用于统一处理日志输出
 */
class LogUtils {
    private static let subsystem = Bundle.main.bundleIdentifier!
    private static let log = OSLog(subsystem: subsystem, category: "PlatformBridge")
    private static let pluginTag = "PlatformBridgePlugin"
    private static let levelError = "ERROR"
    private static let levelWarning = "WARNING"
    
    /**
     * 格式化日志消息
     */
    private static func formatMessage(_ message: String, level: String? = nil) -> String {
        if let level = level {
            return "[\(pluginTag)] \(level): \(message)"
        }
        return "[\(pluginTag)] \(message)"
    }
    
    /**
     * 输出调试日志
     */
    static func debug(_ message: String) {
        let fullMessage = formatMessage(message)
        if #available(iOS 10.0, *) {
            os_log("%{public}@", log: log, type: .debug, fullMessage)
        } else {
            NSLog("%@", fullMessage)
        }
    }
    
    /**
     * 输出信息日志
     */
    static func info(_ message: String) {
        let fullMessage = formatMessage(message)
        if #available(iOS 10.0, *) {
            os_log("%{public}@", log: log, type: .info, fullMessage)
        } else {
            NSLog("%@", fullMessage)
        }
    }
    
    /**
     * 输出错误日志
     */
    static func error(_ message: String) {
        let fullMessage = formatMessage(message, level: levelError)
        if #available(iOS 10.0, *) {
            os_log("%{public}@", log: log, type: .error, fullMessage)
        } else {
            NSLog("%@", fullMessage)
        }
    }
    
    /**
     * 输出警告日志
     */
    static func warning(_ message: String) {
        let fullMessage = formatMessage(message, level: levelWarning)
        if #available(iOS 10.0, *) {
            os_log("%{public}@", log: log, type: .info, fullMessage)  // os_log doesn't have warning level
        } else {
            NSLog("%@", fullMessage)
        }
    }
}

File: lib/log_utils.dart
/// 日志工具类，用于统一处理日志输出
class LogUtils {
  static const String _pluginTag = 'PlatformBridgePlugin';
  static const String _levelError = 'ERROR';
  static const String _levelWarning = 'WARNING';

  /// 格式化日志消息
  static String _formatMessage(String message, [String? level]) {
    if (level != null) {
      return '[$_pluginTag] $level: $message';
    }
    return '[$_pluginTag] $message';
  }

  /// 输出调试日志
  static void debug(String message) {
    _printLog(_formatMessage(message));
  }

  /// 输出信息日志
  static void info(String message) {
    _printLog(_formatMessage(message));
  }

  /// 输出错误日志
  static void error(String message) {
    _printLog(_formatMessage(message, _levelError));
  }

  /// 输出警告日志
  static void warning(String message) {
    _printLog(_formatMessage(message, _levelWarning));
  }

  /// 格式化日志消息并输出
  static void _printLog(String message) {
    // 在调试模式下打印日志
    assert(() {
      // ignore: avoid_print
      print(message);
      return true;
    }());
  }
}
